src/platform/Kconfig

```diff
@@ -532,4 +532,11 @@ config SOF_STACK_SIZE
 	  and might be useful when creating more complex audio processing
 	  components.
 
+config SECONDARY_CORE_DISABLING_TIMEOUT
+	int
+	default 1000
+	depends on MULTICORE
+	help
+	  Count of loops used when waiting for secondary core to enter OFF state.
```

Comment:
maybe milliseconds?

Reply:
@lyakh  Only existing function that I have found with timeout functionality is _**void wait_delay_ms(uint64_t ms)**_. In that use case I don't want to wait the whole time if its not necessary. Loop count with state check was the simplest solution.

Reply:
@tmleman you can use that loop internally, no problem with that. But this Kconfig parameter doesn't explain anything to the user - how long is each loop iteration? You can even literally keep your implementation the same but here don't let the user set a loop count, but a maximum wait time (if that needs to be configurable at all). Make it "maximum wait time in ms" and then calculate your loop count from that.